packages/app-store/nextcloudtalk/api/callback.ts

```diff
@@ -0,0 +1,70 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import { getSafeRedirectUrl } from "@calcom/lib/getSafeRedirectUrl";
+import prisma from "@calcom/prisma";
+
+import getAppKeysFromSlug from "../../_utils/getAppKeysFromSlug";
+import getInstalledAppPath from "../../_utils/getInstalledAppPath";
+import createOAuthAppCredential from "../../_utils/oauth/createOAuthAppCredential";
+import { decodeOAuthState } from "../../_utils/oauth/decodeOAuthState";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const state = decodeOAuthState(req);
+  const userId = req.session?.user.id;
+  if (!userId) {
+    return res.status(404).json({ message: "No user found" });
+  }
+  const { code } = req.query;
+  const appKeys = await getAppKeysFromSlug("nextcloudtalk");
+  const hostUrl = appKeys.nextcloudTalkHost as string;
+  const client_id = appKeys.nextcloudTalkClientId as string;
+  const client_secret = appKeys.nextcloudTalkClientSecret as string;
+
+  const result = await fetch(`${hostUrl}/index.php/apps/oauth2/api/v1/token`, {
```

Comment:
Same as in `VideoApiAdapter.ts`, it might make sense to use `fetchRetry` here instead to deal with HTTP 500 responses on the first call.
```suggestion
  const result = await fetchRetry(`${hostUrl}/index.php/apps/oauth2/api/v1/token`, {
```

Reply:
Let's keep it simple for now; we will follow up if we find this for any other user later